cli/server.go

```diff
@@ -26,11 +31,25 @@ var ServerCmd = cli.Command{
 }
 
 // startEmmyServer configures and starts the gRPC server at the desired port
-func startEmmyServer(port int, certPath, keyPath string) error {
-	srv, err := server.NewProtocolServer(certPath, keyPath)
+func startEmmyServer(port int, certPath, keyPath, logfilePath, loglevel string) error {
+	var err error
+	var logger log.BaseLogger
+
+	if logfilePath == "" {
+		logger, err = log.GetDefaultStdoutLogger("server")
+	} else {
+		logger, err = log.GetDefaultStdoutFileLogger("server", logfilePath)
+	}
 	if err != nil {
 		return err
 	}
+
+	srv, err := server.NewProtocolServer(certPath, keyPath, logger)
+	if err != nil {
+		return err
+	}
+
+	srv.SetLogLevel(loglevel)
```

Comment:
I'm not sure why we're polluting our `Server` with logging functions. I'd rather fully configure the logger and only then assign it to the `Server` struct. Specifically, I'd add following line just after L46:

```go
logger.SetLevel(logLevel)
```

Reply:
I completely agree. Done.